jgrapht-core/src/main/java/org/jgrapht/util/SubSplitQueue.java

```diff
@@ -0,0 +1,177 @@
+    }
+
+    /**
+     * Returns a SubSplitQueue with elements from sortedElements. SortedElements has to a subset of
+     * {0, ..., universeSize - 1}, sorted by desired order, and every entry should be unique
+     * 
+     * @param universeSize the size of the universe
+     * @param sortedElements the elements to be stored (subject to the above-mentioned restrictions)
+     * @return new SubSplitQueue
+     */
+    public static SubSplitQueue subSplitQueueFactory(int universeSize, int[] sortedElements)
+    {
+        return SuperSplitQueue.instantiate(universeSize, sortedElements);
+    }
+
+    /**
+     * Returns a new SubSplitQueue initialized with its index and its parent SuperSplitQueue
+     * 
+     * @param ownIndex index for this queue
+     * @param parent parent
+     */
+    SubSplitQueue(int ownIndex, SuperSplitQueue parent)
+    {
+        this.ownIndex = ownIndex;
+        this.parent = parent;
+    }
+
+    /**
+     * returns its own index
+     * 
+     * @return index
+     */
+    int getOwnIndex()
+    {
+        return ownIndex;
+    }
+
+    /**
+     * Returns the lowest element in the queue without side effects Throws NoSuchElementException if
+     * empty runs in constant time
+     * 
+     * @return the lowest element in the queue
+     */
+    public int peek()
+    {
+        return parent.peek(ownIndex);
+    }
+
+    /**
+     * Returns and removes the lowest element in the queue Throws NoSuchElementException if empty
+     * Runs in constant time
+     * 
+     * @return the lowest element in the queue
+     */
+    public int poll()
+    {
+        return parent.poll(ownIndex);
+    }
+
+    /**
+     * Returns true iff the queue does not contain any elements. Runs in constant time.
+     * 
+     * @return whether it is empty
+     */
+    public boolean isEmpty()
+    {
+        return parent.isEmpty(ownIndex);
+    }
+
+    /**
+     * Removes the given element and keeps the data structure consistent Runs in constant time
+     * 
+     * @param element the element to be removed
+     */
+    public void remove(int element)
+    {
+        parent.remove(element, ownIndex);
+    }
+
+    /**
+     * Splits this queue in two: Elements in splitters are removed and transferred to a new queue,
+     * the others stay. Run time: O(queue size)
+     * 
+     * @param splitters HAVE TO BE SORTED
+     * @return a new SplitQueue holding the split off elements
+     */
+    public SubSplitQueue split(int[] splitters)
+    {
+        return parent.split(splitters, ownIndex);
+    }
+
+    /**
+     * Returns true iff the queue contains element. Runs in constant time.
+     * @param element the element to be checked
+     * @return whether it contains element
+     */
+    public boolean contains(int element)
+    {
+        return parent.contains(element, ownIndex);
+    }
+
+    /**
+     * Iterates over the contained elements Every iteration takes constant time.
```

Comment:
elements. Every

And also... every iteration step takes constant time, and not the whole iteration takes constant time? Or generating an Iterator takes constant time?